File: apps/api/src/application/queries/users/get-user.query.ts
import { IQuery, IQueryHandler, QueryHandler } from '@nestjs/cqrs';
import { Injectable, Inject, NotFoundException } from '@nestjs/common';
import { SupabaseClient, createClient } from '@supabase/supabase-js';
import { ConfigService } from '@nestjs/config';

export class GetUserQuery implements IQuery {
  constructor(public readonly userId: string) {}
}

export interface UserQueryResult {
  id: string;
  email: string;
  nickname: string;
  birthday?: string;
  coupleId?: string;
  createdAt: string;
  updatedAt: string;
}

@Injectable()
@QueryHandler(GetUserQuery)
export class GetUserQueryHandler implements IQueryHandler<GetUserQuery> {
  private supabase: SupabaseClient;

  constructor(private readonly configService: ConfigService) {
    this.supabase = createClient(
      this.configService.get<string>('SUPABASE_URL')!,
      this.configService.get<string>('SUPABASE_SERVICE_KEY')!,
    );
  }

  async execute(query: GetUserQuery): Promise<UserQueryResult> {
    const { data, error } = await this.supabase
      .from('users')
      .select('*')
      .eq('id', query.userId)
      .single();

    if (error || !data) {
      throw new NotFoundException('User not found');
    }

    return {
      id: data.id,
      email: data.email,
      nickname: data.nickname,
      birthday: data.birthday,
      coupleId: data.couple_id,
      createdAt: data.created_at,
      updatedAt: data.updated_at,
    };
  }
}

File: apps/api-kotlin/src/main/kotlin/com/couplecalendar/application/query/user/GetUserQuery.kt
package com.couplecalendar.application.query.user

import com.couplecalendar.application.query.Query
import com.couplecalendar.application.query.QueryHandler
import com.couplecalendar.common.exception.NotFoundException
import com.couplecalendar.infrastructure.persistence.repository.JpaUserRepository
import org.springframework.stereotype.Component
import java.util.UUID

data class GetUserQuery(val userId: UUID) : Query<UserQueryResult>

data class UserQueryResult(
    val id: String,
    val email: String,
    val nickname: String,
    val birthday: String?,
    val coupleId: String?,
    val createdAt: String,
    val updatedAt: String
)

@Component
class GetUserQueryHandler(
    private val jpaUserRepository: JpaUserRepository
) : QueryHandler<GetUserQuery, UserQueryResult> {

    override fun handle(query: GetUserQuery): UserQueryResult {
        val entity = jpaUserRepository.findById(query.userId).orElseThrow {
            NotFoundException("User not found")
        }

        return UserQueryResult(
            id = entity.id.toString(),
            email = entity.email,
            nickname = entity.nickname,
            birthday = entity.birthday?.toString(),
            coupleId = entity.coupleId?.toString(),
            createdAt = entity.createdAt.toString(),
            updatedAt = entity.updatedAt.toString()
        )
    }
}
